File: src/plugin/combat/magic/lunar/teleport/KhazardTeleportSpell.java
package plugin.combat.magic.lunar.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/28/2017
 */
public class KhazardTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 78;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 2, LAW_RUNE, 2, WATER_RUNE, 4);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2656, 3157, 0);
	}
	
	@Override
	public int spellId() {
		return 41;
	}
	
	@Override
	public double exp() {
		return 80;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/KhazardTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/28/2017
 */
class KhazardTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 78
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 2, MagicConstants.LAW_RUNE, 2, MagicConstants.WATER_RUNE, 4)
    }

    override fun destination(): WorldTile {
        return WorldTile(2656, 3157, 0)
    }

    override fun spellId(): Int {
        return 41
    }

    override fun exp(): Double {
        return 80.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}